src/components/WalletPopup/Transactions/index.tsx

```diff
@@ -172,59 +215,101 @@ function ListTransaction({ isMinimal }: { isMinimal: boolean }) {
             </Text>
-          </Flex>
-        ) : (
-          <AutoSizer>
-            {({ height, width }) => (
-              <VariableSizeList
-                height={height}
-                width={width}
-                itemSize={getRowHeight}
-                ref={listRef}
-                outerRef={onRefChange}
-                itemCount={formatTransactions.length}
-                itemData={formatTransactions}
-              >
-                {({ data, index, style }) => (
-                  <RowItem
-                    isMinimal={isMinimal}
-                    style={style}
-                    transaction={data[index]}
-                    index={index}
-                    key={data[index].hash}
-                    setRowHeight={setRowHeight}
-                    cancellingOrderInfo={cancellingOrderInfo}
-                  />
-                )}
-              </VariableSizeList>
-            )}
-          </AutoSizer>
+          </Row>
+          <Row gap="16px" flexDirection={upToExtraSmall ? 'column' : 'row'}>
+            <ButtonOutlined onClick={onCloseClearTxModal}>{t`Cancel`}</ButtonOutlined>
+            <ButtonPrimary onClick={onClearAllTransactions}>{t`Clear All`}</ButtonPrimary>
+          </Row>
+        </ClearTxWrapper>
+      </Modal>
+      <Wrapper>
+        <Tab<TRANSACTION_GROUP | string> activeTab={activeTab} setActiveTab={setActiveTab} tabs={filterTab} />
+        <ContentWrapper>
+          {formatTransactions.length === 0 ? (
+            <Flex flexDirection="column" alignItems="center" color={theme.subText} sx={{ gap: 10, marginTop: '20px' }}>
+              <Info size={32} />
+              <Text fontSize={'14px'}>
+                <Trans>You have no Transaction History.</Trans>
+              </Text>
+            </Flex>
+          ) : (
+            <AutoSizer>
+              {({ height, width }) => (
+                <VariableSizeList
+                  height={height}
+                  width={width}
+                  itemSize={getRowHeight}
+                  ref={listRef}
+                  outerRef={onRefChange}
+                  itemCount={formatTransactions.length}
+                  itemData={formatTransactions}
+                >
+                  {({ data, index, style }) => (
+                    <RowItem
+                      isMinimal={isMinimal}
+                      style={style}
+                      transaction={data[index]}
+                      index={index}
+                      key={data[index].hash}
+                      setRowHeight={setRowHeight}
+                      cancellingOrderInfo={cancellingOrderInfo}
+                    />
+                  )}
+                </VariableSizeList>
+              )}
+            </AutoSizer>
+          )}
+        </ContentWrapper>
+        {formatTransactions.length !== 0 && (
```

Comment:
since we only remove pending tx, I think should change this condition to pendingTxs.length